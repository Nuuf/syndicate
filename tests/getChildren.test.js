/* eslint-disable @typescript-eslint/no-var-requires */
const syndicate = require('../dist');

test('gets the children or a slice of a parent', () => {
  const root = syndicate.createRoot();
  const parent = syndicate.add(root, syndicate.create({}, 'parent'));
  syndicate.add(root, syndicate.create({}, 'A'), parent);
  syndicate.add(root, syndicate.create({}, 'B'), parent);
  syndicate.add(root, syndicate.create({}, 'C'), parent);
  syndicate.add(root, syndicate.create({}, 'D'), parent);
  syndicate.add(root, syndicate.create({}, 'E'), parent);
  const children = syndicate.getChildren(root, parent, 2, 0);
  expect(JSON.stringify(children, null, 2)).toMatchSnapshot();
});
