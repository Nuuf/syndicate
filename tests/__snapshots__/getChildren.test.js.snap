// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gets the children or a slice of a parent 1`] = `
"[
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":1,\\\\\\"parentId\\\\\\":\\\\\\"parent\\\\\\",\\\\\\"index\\\\\\":2,\\\\\\"id\\\\\\":\\\\\\"A\\\\\\",\\\\\\"data\\\\\\":{},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":1,\\\\\\"parentId\\\\\\":\\\\\\"parent\\\\\\",\\\\\\"index\\\\\\":3,\\\\\\"id\\\\\\":\\\\\\"B\\\\\\",\\\\\\"data\\\\\\":{},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\"
]"
`;
